x/tx/signing/handler_map.go

```diff
@@ -11,17 +11,21 @@ import (
 // based on sign mode.
 type HandlerMap struct {
 	signModeHandlers map[signingv1beta1.SignMode]SignModeHandler
+	defaultMode      signingv1beta1.SignMode
 	modes            []signingv1beta1.SignMode
 }
 
-// NewHandlerMap constructs a new sign mode handler map.
+// NewHandlerMap constructs a new sign mode handler map. The first handler is used as the default.
 func NewHandlerMap(handlers ...SignModeHandler) *HandlerMap {
 	res := &HandlerMap{
 		signModeHandlers: map[signingv1beta1.SignMode]SignModeHandler{},
 	}
 
-	for _, handler := range handlers {
+	for i, handler := range handlers {
 		mode := handler.Mode()
+		if i == 0 {
+			res.defaultMode = mode
```

Comment:
A unit test for this would be good, even though this is likely tested indirectly in many places.

It would also be good to have test coverage defining what should happen if `NewHandlerMap` is called without any handlers. I'm guessing a panic would be appropriate.